self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryInvoices.java

```diff
@@ -193,11 +190,6 @@ public void pay(final int id) {
          */
         private final Supplier<Stream<Invoice>> contractInvoices;
 
-        /**
-         * Current active invoice.
-         */
-        private Integer activeId;
```

Comment:
@criske Why remove this?

Reply:
Was a bug with that aproach.
Will fail the test when we check if the invoice contract is paid. You have to check the actual storage if payment is done.
Here is the corner case :
`Invoices invoicesOfContract = contract.invoices() .ofContract(contractId);`
then we add contract task through overall invoices not through **invoicesOfContract**:
`storage.invoices().add(taskOfContract, duration)`
then we "pay"
`storage.invoices().pay(contractInvoiceId)`
Now if you call  `invoicesOfContract.isPaid(contractInvoiceId) ` the old code will return **false** instead of **true** (stale state of **activeId**).